platform/ios/src/MGLMapView.mm

```diff
@@ -2932,17 +2929,34 @@ - (void)selectAnnotation:(id <MGLAnnotation>)annotation animated:(BOOL)animated
     if (annotationTag == MGLAnnotationTagNotFound && annotation != self.userLocation)
     {
         [self addAnnotation:annotation];
+        annotationTag = [self annotationTagForAnnotation:annotation];
+        if (annotationTag == MGLAnnotationTagNotFound) return;
     }
     
-    // The annotation can’t be selected if no part of it is hittable.
-    CGRect positioningRect = [self positioningRectForCalloutForAnnotationWithTag:annotationTag];
+    MGLAnnotationContext &annotationContext = _annotationContextsByAnnotationTag.at(annotationTag);
+    CGRect positioningRect;
+   
+    if (annotationContext.annotationView)
+    {
+        // Annotations represented by views use the view frame as the positioning rect.
+        positioningRect = annotationContext.annotationView.frame;
+        positioningRect.origin.y -= 0.5;
+    }
+    else
+    {
+        // Annotations represented by GL images use the image frame as the positioning rect.
+        positioningRect = [self positioningRectForCalloutForAnnotationWithTag:annotationTag];
+    }
+    
+     // The client can request that any annotation be selected (even ones that are offscreen).
+     // The annotation can’t be selected if no part of it is hittable.
     if ( ! CGRectIntersectsRect(positioningRect, self.bounds) && annotation != self.userLocation)
     {
         return;
     }
 
     self.selectedAnnotation = annotation;
-
+  
```

Comment:
Nit: stray whitespace. (I’m fine with whitespace on blank lines, but it should line up with the surrounding code.)